core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -2169,6 +2170,21 @@ protected static Calendar getCalendarNotTooNear(int timeUnit) {
         "VARCHAR(33335) NOT NULL");
     tester.checkNull("x'ff' || cast(null as varbinary)");
     tester.checkNull(" cast(null as ANY) || cast(null as ANY) ");
+    tester.checkString("cast('a' as varchar) || cast('b' as varchar) "
+        + "|| cast('c' as varchar)", "abc", "VARCHAR NOT NULL");
+
+    final SqlTester tester1 = tester(SqlLibrary.MYSQL);
+    final SqlTester tester2 = tester(SqlLibrary.POSTGRESQL);
+    final SqlTester tester3 = tester(SqlLibrary.ORACLE);
```

Comment:
oracle `CONCAT (char1, char2)` only accept how to reflect two parameters?

Reply:
I don't think that it is a serious problem since concat function will be translated to || operator and it can work well for oracle. In a practical application scenario, we may use two SqlLibaray at a same time, if they have different concat function, ambiguity will be caused when check arguements, even we register oracle concat as a alias of SqlStdOperatorTable.CONCAT.

Reply:
> I don't think that it is a serious problem since concat function will be translated to || operator and it can work well for oracle and also support concat('a', 'b'). In a practical application scenario, we may use two SqlLibaray at a same time, if they have different concat function, ambiguity will be caused when check arguements, even we register oracle concat as a alias of SqlStdOperatorTable.CONCAT.

I did not say that this is a serious problem, but I noticed that oracle's `concat` is a special case of your `concat` implementation. Because oracle `concat` only supports two parameters.